File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/UpsertOrderDto.java
package com.sudo248.orderservice.controller.order.dto;

import com.sudo248.orderservice.repository.entity.order.OrderStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpsertOrderDto {
    private String orderId;
    private String cartId;
    private String paymentId;
    private String promotionId;
    private LocalDateTime createdAt;
    private OrderStatus status = OrderStatus.PREPARE;
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/order/UpsertOrderDto.kt
package com.sudo248.sudoo.data.dto.order

import com.sudo248.sudoo.domain.entity.order.OrderStatus

data class UpsertOrderDto(
    val orderId: String? = null,
    val cartId: String,
    val paymentId: String? = null,
    val promotionId: String? = null,
    val status: OrderStatus? = null,
)